common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -210,8 +252,23 @@ private AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
       File metaFile =
         appShuffleInfo.getMergedShuffleMetaFile(shuffleId, shuffleMergeId, reduceId);
       try {
-        return newAppShufflePartitionInfo(appShuffleInfo.appId, shuffleId, shuffleMergeId,
-          reduceId, dataFile, indexFile, metaFile);
+        AppShufflePartitionInfo appShufflePartitionInfo =
+            newAppShufflePartitionInfo(appShuffleInfo.appId, appShuffleInfo.attemptId,
+            shuffleId, shuffleMergeId, reduceId, dataFile, indexFile, metaFile);
+        if (db != null) {
+          try{
+            byte[] dbKey =
+                dbAppAttemptShufflePartitionKeyWithCustomPrefix(
+                  appShufflePartitionInfo.appShufflePartitionId,
+                  APP_ATTEMPT_SHUFFLE_PARTITION_KEY_PREFIX);
+            // We can reload a partition's state solely from the partition ID. Thus only the
+            // partition Ids are stored in levelDB.
+            db.put(dbKey, new byte[0]);
+          } catch (Exception e) {
+            logger.error("Error saving active app shuffle partition", e);
+          }
+        }
```

Comment:
We have a bunch of 
```
if (db != null) { 
  try {
    byte[] key = generate key
    db.put(key, value)
  } catch (Exception ex) {
    log message
  }
}
```
sprinkled across this class.

Refactor these and move them to util methods ? So that the main flow is clearer ?

```suggestion
  writeShufflePartitionToDb(appShufflePartitionInfo.appShufflePartitionId)
```